pkgs/applications/virtualization/xen/generic/default.nix

```diff
@@ -381,7 +384,8 @@ stdenv.mkDerivation (finalAttrs: {
     ]
     ++ lib.lists.optional withInternalOVMF nasm
     ++ lib.lists.optional withFlask checkpolicy
-    ++ lib.lists.optional (lib.strings.versionOlder version "4.19") systemdMinimal;
+    ++ lib.lists.optional (lib.strings.versionOlder version "4.19") systemdMinimal
+    ++ lib.lists.optional stdenv.targetPlatform.isAarch64 dtc;
```

Comment:
`targetPlatform` only applies to compilers. You want either `stdenv.hostPlatform`, which is the platform the resulting package will run on, or `stdenv.buildPlatform`, the platform the build is running on. Probably the former.